aclk/mqtt.c

```diff
@@ -322,6 +322,7 @@ int _link_send_message(char *topic, unsigned char *message, int *mid)
         return rc;
 
     int msg_len = strlen((char*)message);
+    errno = 0;
```

Comment:
Why are you overwriting `errno`?

Reply:
It's a good practice whenever you call `error()` but you don't want it to automatically print `strerror(errno)` of a previous unrelated error.